Enable Gtk+GStreamer gles2 support again

ystreet opened the pull request:
https://phabricator.endlessm.com/T24268

ystreet committed:
manifest: gst-plugins-base: explicitly enable gles2 support for arm

Otherwise, if gles2 is enabled or not is entirely dependant on whether
compatible OpenGL and GLES2 headers are in the runtime.

https://phabricator.endlessm.com/T24268

ystreet committed:
manifest: gst-plugins-good: enable EGL/X11 support for gtkglsink

https://gitlab.freedesktop.org/gstreamer/gst-plugins-good/merge_requests/134

https://phabricator.endlessm.com/T24268

dsd commented on generate-manifest.py:
```diff
@@ -136,12 +137,6 @@ def edit_manifest(data, arch, branch, runtime_version):
                 gtk_module['sources'].append({ 'type': 'patch', 'path': patch })
             data['modules'].insert(0, gtk_module)
             break
-        if m['name'] == 'gstreamer-plugins-good':
```
Removing this section and moving it to the ARM-specific section below means that `gtkgstwidget-add-ready-to-show-signal.patch` is no longer applied on x86_64 builds. Is that intentional & desired?

ystreet replied:
Hrm.  I moved it there as otherwise the patches were being applied to gst 1.10 instead of gst 1.14.  What is this 'ready-to-show' signal patch for exactly?

dsd replied:
I'm also new to this codebase, but https://github.com/endlessm/endless-sdk-flatpak/commit/bc7341db8ed8ad2c1521b66a1258aa186cf8be88 has the background there.

ystreet replied:
That still doesn't explain why you need that patch?  What is the user of the signal trying to do?  Why doesn't preroll help here?

ptomato replied:
Here is more background: https://phabricator.endlessm.com/T24812

ystreet replied:
Ok, then these days gtkglsink will draw black if there are no buffers to render instead of whatever may or may not be in a texture.  I assume that fixes this 'white glitch' frame/s you may have been seeing.

dsd replied:
That's good to know, however it looks like this codebase is still using gstreamer-1.14 (from freedesktop SDK 1.6) so probably the best course of action is to leave this patch in place for now.

ystreet replied:
That 'draw black' code has been there since 1.6 so I still don't see why it's necessary :)

dsd replied:
When the issue was filed and resolved, gstreamer-1.10 was being used (and still is to this date). So please err on the side of this patch being necessary, even if we don't really know why.

ystreet replied:
Huh, the patch was never actually applied.  The for loop would break before ever applying the 'gtk: add ready-to-show' patch.

dsd replied:
Oh, well spotted. Those "break" statements will prevent the other module from being handled. There's something to fix there.

As for which module is being handled in current builds (and breaking before the other gets handled), I'm not sure. Python's dictionary type is (potentially) unordered, so even though gtk3 appears first in the manifest, it may not definitively be the one processed first in python.

(As of python3.6 it should obey the manifest order though - https://mail.python.org/pipermail/python-dev/2016-September/146327.html - so I wonder which python version is running on the build host here)

If the patch was never applied, then maybe it makes sense to drop it completely. But I'm not exactly certain that this was the case.

@xjuan @erikos  any suggestions what to do here? Is there a way that we can verify that gtkgstwidget-add-ready-to-show-signal.patch was actually included in the published builds? If it was accidently omitted, do you actually want it or can it be dropped?

cosimoc replied:
@dsd The easiest way to verify whether the patch was applied is to look for the signal on the widget, like this code does: https://github.com/endlessm/hack-toolbox-app/blob/master/src/playbin.js#L90
I remember @xjuan spending a lot of time getting the patch right and testing it, so I'd be surprised if it was not needed/applied after all.

dsd replied:
I'm a bit pressed for time to figure out how to do that.

@ystreet please proceed to assume the patch is still required, and fix the break statements so that the patch is actually applied, as well as making the required changes for ARM gles2.

ystreet committed:
generate-manifest: ensure that x86-64 includes the necessary gst patches

Also ensuring that the SDK uses gst 1.14 from the GNOME module like the
rest of the SDK.
